frontend/src/stores/redux/loggedUser/loggedUserSlice.ts

```diff
@@ -0,0 +1,41 @@
+import {createSlice, PayloadAction} from '@reduxjs/toolkit'
+import {RootState} from "../store";
+import {ILoggedUser} from "../../../types/loggedUser";
+import {UserType} from "../../../types/UserType";
+
+const initialState: ILoggedUser = {
+    id: -1,
+    email: "",
+    username: "",
+    userType: UserType.Customer,
+    token: ""
+}
+
+export const loggedUserSlice = createSlice({
+    name: 'loggedUser',
+    initialState,
+    reducers: {
+        signIn: (state, action: PayloadAction<ILoggedUser>) => {
+            state.id = action.payload.id;
+            state.email = action.payload.email;
+            state.username = action.payload.username;
+            state.userType = action.payload.userType;
+            state.token = action.payload.token;
+        },
+        logOut: state => {
+            state.id = -1;
+            state.email = '';
+            state.username = '';
+            state.userType = UserType.Customer;
+            state.token = '';
+        }
```

Comment:
Firstly as I said in previous comment, use destructuring, spread, rest syntax to faster copy data. https://www.digitalocean.com/community/tutorials/understanding-destructuring-rest-parameters-and-spread-syntax-in-javascript

Secondly, I think you don't need to change current state. As I know it's a bad practice, you have to return new state based on old state and new data and then return final result. Since we don't need old state, we can return just new data 
```typescript
reducers: {
        signIn: (state, action: PayloadAction<ILoggedUser>) => {
            return {...action.payload}
        },
        logOut: state => {
            return {...initialState}
        }
    }
```